xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-ui/src/main/resources/AppWithinMinutes/Content.xml

```diff
@@ -37,49 +37,51 @@
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
   <content>{{velocity}}
-#set ($className = $object.getxWikiClass().name)
-#if ($doc.fullName == $className)
-  ## We are editing the class so the content must be read from / written to the template document.
-  #set ($name = 'templateContent')
-  #set ($document = $xwiki.getDocument("$stringtool.removeEnd($className, 'Class')Template"))
-  ## Don't load the WYSIWYG editor when editing the class, because it's too heavy.
-  #set ($useWysiwygEditor = false)
-#else
-  ## We are editing an application entry so the content must be read from / written to the current document.
-  #set ($name = 'content')
-  #set ($document = $tdoc)
-  ## Use the preferred content editor.
-  #set ($useWysiwygEditor = $xwiki.getUserPreference('editor') == 'Wysiwyg')
-#end
-#if ($type == 'edit')
-  {{html clean="false"}}
-  ## The "content" id is expected by some JavaScript and CSS code.
-  #set ($id = 'content')
-  #if (!$useWysiwygEditor)
-    &lt;div id="xwikieditcontentinner"&gt;
-      ## The tool bar may have an entry to insert an HTML macro. Make sure it doesn't break the HTML macro we are currently in.
-      #set ($toolBar = "#template('simpleedittoolbar.vm')")
-      $!toolBar.replace('{{', '&amp;#123;&amp;#123;')
-      ## Display a simple textarea.
-      &lt;textarea id="$id" cols="80" rows="25" name="$name"&gt;$escapetool.xml($document.content)&lt;/textarea&gt;
-  #end
-  #if ($useWysiwygEditor)
-    $!services.edit.syntaxContent.wysiwyg($document.content, $document.syntax, {
-      'id': "$id",
-      'name': "$name",
-      'rows': 25,
-      'cols': 80,
-      'full': true,
-      'restricted': $document.isRestricted()
-    })
+#if ($type == 'edit' || $type == 'hidden')
+  #set ($className = $object.getxWikiClass().name)
+  #if ($doc.fullName == $className)
+    ## We are editing the class so the content must be read from / written to the template document.
+    #set ($name = 'templateContent')
+    #set ($editedDocument = $xwiki.getDocument("$stringtool.removeEnd($className, 'Class')Template"))
+    ## Don't load the WYSIWYG editor when editing the class, because it's too heavy.
+    #set ($useWysiwygEditor = false)
   #else
-    &lt;/div&gt;
+    ## We are editing an application entry so the content must be read from / written to the current document.
+    #set ($name = 'content')
+    #set ($editedDocument = $tdoc)
+    ## Use the preferred content editor.
+    #set ($useWysiwygEditor = $xwiki.getUserPreference('editor') == 'Wysiwyg')
+  #end
+  #if ($type == 'edit')
+    {{html clean="false"}}
```

Comment:
We can have the HTML macro a single time, around the check for type, so that we don't duplicate this line with line 81 below. Same for lines 79 and 83.

Same for title.

Reply:
Done